crates/secrets/src/model/secret_rule.rs

```diff
@@ -3,10 +3,129 @@
+            SecretRuleMatchValidationHttpMethod::Delete => HttpMethod::Delete,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct SecretRuleMatchValidation {
+    #[serde(rename = "type")]
+    pub r#type: String,
+    pub endpoint: Option<String>,
+    pub hosts: Option<Vec<String>>,
+    pub request_headers: Option<HashMap<String, String>>,
+    pub http_method: Option<SecretRuleMatchValidationHttpMethod>,
+    pub timeout_seconds: Option<u64>,
+    pub valid_http_status_code: Option<Vec<SecretRuleMatchValidationHttpCode>>,
+    pub invalid_http_status_code: Option<Vec<SecretRuleMatchValidationHttpCode>>,
+}
+
+impl SecretRuleMatchValidation {
+    pub fn get_request_headers(&self) -> Vec<RequestHeader> {
+        if let Some(rhs) = &self.request_headers {
+            rhs.iter()
+                .map(|(k, v)| RequestHeader {
+                    key: k.clone(),
+                    value: v.clone(),
+                })
+                .collect()
+        } else {
+            Vec::new()
+        }
+    }
+}
+
+impl TryFrom<SecretRuleMatchValidation> for MatchValidationType {
+    type Error = &'static str;
+
+    fn try_from(value: SecretRuleMatchValidation) -> Result<Self, Self::Error> {
+        match value.r#type.as_str() {
+            AWS_ID_STRING => Ok(MatchValidationType::Aws(AwsId)),
+            AWS_SECRET_STRING => Ok(MatchValidationType::Aws(AwsSecret(AwsConfig::default()))),
+            AWS_SESSION_STRING => Ok(MatchValidationType::Aws(AwsSession)),
+            CUSTOM_HTTP_STRING => {
+                let invalid_ports: Vec<Range<u16>> = value
+                    .invalid_http_status_code
+                    .clone()
+                    .unwrap_or_default()
+                    .iter()
+                    .map(|v| Range {
+                        start: v.start,
+                        end: v.end,
+                    })
+                    .collect();
+                let valid_ports: Vec<Range<u16>> = value
+                    .valid_http_status_code
+                    .clone()
+                    .unwrap_or_default()
+                    .iter()
+                    .map(|v| Range {
+                        start: v.start,
+                        end: v.end,
+                    })
+                    .collect();
+                Ok(MatchValidationType::CustomHttp(
+                    HttpValidatorConfigBuilder::new(value.endpoint.clone().unwrap())
+                        .set_hosts(value.hosts.clone().unwrap_or_default())
+                        .set_invalid_http_status_code(invalid_ports)
+                        .set_request_header(value.clone().get_request_headers())
+                        .set_valid_http_status_code(valid_ports)
+                        .set_method(value.http_method.unwrap().into())
+                        .build()
+                        .unwrap(),
+                ))
+            }
+
+            _ => Err("invalid type"),
+        }
+    }
```

Comment:
We're cloning in spots we don't need to, because you're using part of an Option that isn't used elsewhere, so why not just `take()` it?

```suggestion
    fn try_from(mut value: SecretRuleMatchValidation) -> Result<Self, Self::Error> {
        match value.r#type.as_str() {
            AWS_ID_STRING => Ok(MatchValidationType::Aws(AwsId)),
            AWS_SECRET_STRING => Ok(MatchValidationType::Aws(AwsSecret(AwsConfig::default()))),
            AWS_SESSION_STRING => Ok(MatchValidationType::Aws(AwsSession)),
            CUSTOM_HTTP_STRING => {
                let invalid_ports: Vec<Range<u16>> = value
                    .invalid_http_status_code
                    .take()
                    .unwrap_or_default()
                    .iter()
                    .map(|v| Range {
                        start: v.start,
                        end: v.end,
                    })
                    .collect();
                let valid_ports: Vec<Range<u16>> = value
                    .valid_http_status_code
                    .take()
                    .unwrap_or_default()
                    .iter()
                    .map(|v| Range {
                        start: v.start,
                        end: v.end,
                    })
                    .collect();

                Ok(MatchValidationType::CustomHttp(
                    HttpValidatorConfigBuilder::new(value.endpoint.clone().unwrap())
                        .set_hosts(value.hosts.take().unwrap_or_default())
                        .set_invalid_http_status_code(invalid_ports)
                        .set_request_header(value.get_request_headers())
                        .set_valid_http_status_code(valid_ports)
                        .set_method(value.http_method.unwrap().into())
                        .build()
                        .unwrap(),
                ))
            }

            _ => Err("invalid type"),
        }
    }
```